jenkins/test/testsuite/Jenkinsfile

```diff
@@ -68,20 +70,46 @@ pipeline {
             post() {
                 always {
                     script {
-                        publish()
+                        publishResults()
                     }
                     cleanWs disableDeferredWipeout: true, deleteDirs: true
                 }
             }
         }
     }
+    post() {
+        success {
+            node('Jenkins-Agent-al2-x64-m5xlarge') {
+                script {
+                    def stashed = lib.jenkins.Messages.new(this).get(['Integ-Test'])
+                    publishNotification(":white_check_mark:", "Successful Build", "\n${stashed}")
+                }
+            }
+        }
+        failure {
+            node('Jenkins-Agent-al2-x64-m5xlarge') {
+                script  {
+                    def stashed = lib.jenkins.Messages.new(this).get(['Integ-Test'])
+                    publishNotification(":warning:", "Failed Build", "\n${stashed}")
+                }
+            }
+        }
+    }
 }
 
 /** Publishes test results to S3 **/
-void publish() {
+void publishResults() {
     def artifactPath = "bundle-build/${opensearch_version}/${build_id}/${platform}/${architecture}"
     withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
         s3Upload(file: 'test-results', bucket: "${ARTIFACT_BUCKET_NAME}", path: "${artifactPath}/test-results")
     }
-    echo "Uploaded Test Results to ${artifactPath}/test-results"
+    echo "Uploaded Test Results to s3://${ARTIFACT_BUCKET_NAME}/${artifactPath}/test-results"
+    lib.jenkins.Messages.new(this).add("${STAGE_NAME}", "s3://${ARTIFACT_BUCKET_NAME}/${artifactPath}/test-results")
+}
+
+/** Publishes a notification to a slack instance*/
```

Comment:
Same as above. We have https://github.com/opensearch-project/opensearch-build/blob/main/vars/publishNotification.groovy. Parameterize the `credentialsId` and pass in `INTEG_TEST_WEBHOOK` vs. `BUILD_NOTICE_WEBHOOK`.